system/lib/libcxxabi/src/cxa_guard.cpp

```diff
@@ -9,6 +9,29 @@
 
 #include "abort_message.h"
 
+#if __EMSCRIPTEN__
+
+// XXX EMSCRIPTEN making guard operations simple and LTO-optimizable opens up
```

Comment:
Isn't this making the guards thread-unsafe, and that would break USE_PTHREADS-enabled compilation?